### PR TITLE
add checkstyle

### PR DESCRIPTION
**Pull request by ktalanda**
addresses 
- https://github.com/rspective/voucherify-android-sdk/issues/26
- https://trello.com/c/rX0xgN9i/546-android-sdk-add-ci

implementation based on http://vincentbrison.com/2014/07/19/how-to-improve-quality-and-syntax-of-your-android-code/

rules based on square solution: https://github.com/square/register-android-sdk/blob/master/checkstyle.xml

to invoke checkstyle run `./gradlew checkstyle` or `./gradlew check`

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        <module name="AvoidStarImport"/>
+        <module name="IllegalImport"/>
+        <module name="RedundantImport"/>
+        <module name="UnusedImports"/>
+
+
+        <!-- Checks for Size Violations.                    -->
+        <!-- See http://checkstyle.sf.net/config_sizes.html -->
+        <module name="LineLength">
+            <property name="max" value="120"/>
```
Comment (polok) [inline config/checkstyle/checkstyle.xml:67]: @ktalanda Maybe we could increase this length to for instance 160 characters. What do you think?

Comment (ktalanda) [reply config/checkstyle/checkstyle.xml:67]: 160... pretty long :) I would prefer 120. Easier to browse... But don't mind to go for 160 if you have any particular reason.

**Comment by ktalanda** [comment]
blocked by https://github.com/voucherifyio/voucherify-android-sdk/pull/38

**Comment by polok** [comment]
@ktalanda Please, resolve conflicts

**Review by polok — APPROVED** [review]
👍

**Pull request merged**